pandora-server-directory/src/account/accountDirectMessages.ts

```diff
@@ -36,6 +32,27 @@ export class AccountDirectMessages {
 		this._dms = data ?? [];
 	}
 
+	public async getDirectMessageInfo(): Promise<IDirectoryDirectMessageInfo[]> {
```

Comment:
This method should be async synchronized to avoid extra queries while one is already pending.